test/governance/StkTruToken.test.ts

```diff
@@ -0,0 +1,175 @@
+    await stkToken.initialize(trustToken.address, tusd.address, oracle.address)
+
+    await trustToken.mint(owner.address, amount)
+    await trustToken.approve(stkToken.address, amount)
+  })
+
+  describe('Oracle', () => {
+    it('only owner can change oracle', async () => {
+      await expect(stkToken.connect(staker).setOracle(staker.address)).to.be.revertedWith('only owner')
+    })
+
+    it('emits event', async () => {
+      await expect(stkToken.setOracle(staker.address)).to.emit(stkToken, 'OracleChanged')
+        .withArgs(staker.address)
+      expect(await stkToken.oracle()).to.equal(staker.address)
+    })
+  })
+
+  describe('Staking-Unstaking', () => {
+    it('stake emits event', async () => {
+      await expect(stkToken.stake(amount)).to.emit(stkToken, 'Stake').withArgs(owner.address, amount, amount)
+    })
+
+    it('unstake emits event', async () => {
+      await stkToken.stake(amount)
+      await timeTravel(provider, stakeCooldown)
+      await expect(stkToken.unstake(amount)).to.emit(stkToken, 'Unstake').withArgs(owner.address, amount, amount, 0)
+    })
+
+    it('tokens are burnt on unstake', async () => {
+      await stkToken.stake(amount)
+      await timeTravel(provider, stakeCooldown)
+      await stkToken.unstake(amount)
+      expect(await stkToken.totalSupply()).to.equal(0)
+    })
+
+    it('cannot unstake before cooldown has passed', async () => {
+      await stkToken.stake(amount)
+      await timeTravel(provider, stakeCooldown - DAY)
+      await expect(stkToken.unstake(amount)).to.be.revertedWith('StkTruToken: Stake is locked')
+    })
+
+    describe('single user', () => {
+      it('stakes, unstakes, gets same amount of TRU', async () => {
+        await stkToken.stake(amount)
+        await timeTravel(provider, stakeCooldown)
+        await stkToken.unstake(amount)
+        expect(await trustToken.balanceOf(owner.address)).to.equal(amount)
+      })
+
+      it('stakes, then some TRU and TUSD are transferred to stake, then unstake', async () => {
+        await stkToken.stake(amount)
+        await trustToken.mint(stkToken.address, parseTRU(10))
+        await tusd.mint(stkToken.address, parseEth(10))
+        await timeTravel(provider, stakeCooldown)
+        await stkToken.unstake(amount)
+        expect(await trustToken.balanceOf(owner.address)).to.equal(amount.add(parseTRU(10)))
+        expect(await tusd.balanceOf(owner.address)).to.equal(parseEth(10))
+      })
+
+      it('some TRU and TUSD are transferred to stake, then stake & unstake', async () => {
+        await trustToken.mint(stkToken.address, parseTRU(10))
+        await tusd.mint(stkToken.address, parseEth(10))
+        await stkToken.stake(amount)
+        await timeTravel(provider, stakeCooldown)
+        await stkToken.unstake(amount)
+        expect(await trustToken.balanceOf(owner.address)).to.equal(amount.add(parseTRU(10)))
+        expect(await tusd.balanceOf(owner.address)).to.equal(parseEth(10))
+      })
+    })
+
```

Comment:
Maybe we should also make a test, that checks if cooldown works properly.
User stakes -> cooldown passes -> stakes again and immediately unstakes.
That should revert.